pvlib/pvarray.py

```diff
@@ -0,0 +1,220 @@
+"""
+This module contains implementations of PV module and array electrical models.
+
+These models are used to predict the electrical behavior of pv modules
+or collections of pv modules (arrays).  The primary inputs are
+effective irradiance and operating temperature and the outputs may range from
+power or efficiency at the maximum power point to complete IV curves.
+Supporting functions and parameter fitting functions may also be found here.
+"""
+
+import numpy as np
+from scipy.optimize import curve_fit
+from scipy.special import exp10
+
+
+def pvefficiency_adr(effective_irradiance, temp_cell,
+                     k_a, k_d, tc_d, k_rs, k_rsh):
+    '''
+    Calculate PV module efficiency using the ADR model.
+
+    The efficiency varies with irradiance and operating temperature
+    and is determined by 5 model parameters as described in [1]_.
+
+    Parameters
+    ----------
+    effective_irradiance : numeric, non-negative
+        The effective irradiance incident on the PV module. [W/m²]
+
+    temp_cell : numeric
+        The PV module operating temperature. [°C]
+
+    k_a : numeric
+        Absolute scaling factor, which is equal to the efficiency at
+        reference conditions. This factor allows the model to be used
+        with relative or absolute efficiencies, and to accommodate data sets
+        which are not perfectly normalized but have a slight bias at
+        the reference conditions. [unitless|%]
+
+    k_d : numeric, negative
+        “Dark irradiance” or diode coefficient which influences the voltage
+        increase with irradiance. [unitless]
+
+    tc_d : numeric
+        Temperature coefficient of the diode coefficient, which indirectly
+        influences voltage. Because it is the only temperature coefficient
+        in the model, its value will also reflect secondary temperature
+        dependencies that are present in the PV module. [unitless]
+
+    k_rs and k_rsh : numeric
+        Series and shunt resistance loss factors. Because of the normalization
+        they can be read as power loss fractions at reference conditions.
+        For example, if k_rs is 0.05, the internal loss assigned to the
+        series resistance has a magnitude equal to 5% of the module output.
+        [unitless]
+
+    Returns
+    -------
+    eta : numeric
+        The efficiency of the module at the specified irradiance and
+        temperature.
+
+    Notes
+    -----
+    The efficiency values may be absolute or relative, and may be expressed
```

Comment:
"efficiency values" refers to `eta`? It would help to specify.  Use ``k_a`` in this paragraph.